presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java

```diff
@@ -55,112 +53,19 @@ public static Block repeat(
-    @SqlType("array(T)")
-    public static Block repeat(
-            @TypeParameter("T") Type type,
-            @SqlNullable @SqlType("T") Long element,
-            @SqlType(StandardTypes.INTEGER) long count)
-    {
-        BlockBuilder blockBuilder = createBlockBuilder(type, count);
-        if (element == null) {
-            return repeatNullValues(blockBuilder, count);
-        }
-        for (int i = 0; i < count; i++) {
-            type.writeLong(blockBuilder, element);
-        }
-        return blockBuilder.build();
-    }
-
-    @TypeParameter("T")
-    @SqlType("array(T)")
-    public static Block repeat(
-            @TypeParameter("T") Type type,
-            @SqlNullable @SqlType("T") Slice element,
-            @SqlType(StandardTypes.INTEGER) long count)
-    {
-        BlockBuilder blockBuilder = createBlockBuilder(type, count);
-        if (element == null) {
-            return repeatNullValues(blockBuilder, count);
-        }
-        if (count > 0) {
-            type.writeSlice(blockBuilder, element);
-            checkMaxSize(blockBuilder.getSizeInBytes(), count);
-        }
-        for (int i = 1; i < count; i++) {
-            type.writeSlice(blockBuilder, element);
-        }
-        return blockBuilder.build();
-    }
-
-    @TypeParameter("T")
-    @SqlType("array(T)")
-    public static Block repeat(
-            @TypeParameter("T") Type type,
-            @SqlNullable @SqlType("T") Boolean element,
-            @SqlType(StandardTypes.INTEGER) long count)
-    {
-        BlockBuilder blockBuilder = createBlockBuilder(type, count);
-        if (element == null) {
-            return repeatNullValues(blockBuilder, count);
-        }
-        for (int i = 0; i < count; i++) {
-            type.writeBoolean(blockBuilder, element);
-        }
-        return blockBuilder.build();
-    }
-
-    @TypeParameter("T")
-    @SqlType("array(T)")
-    public static Block repeat(
-            @TypeParameter("T") Type type,
-            @SqlNullable @SqlType("T") Double element,
-            @SqlType(StandardTypes.INTEGER) long count)
-    {
-        BlockBuilder blockBuilder = createBlockBuilder(type, count);
-        if (element == null) {
-            return repeatNullValues(blockBuilder, count);
-        }
-        for (int i = 0; i < count; i++) {
-            type.writeDouble(blockBuilder, element);
-        }
-        return blockBuilder.build();
+        Block result = RunLengthEncodedBlock.create(type, element, (int) count);
+        checkCondition(result.getSizeInBytes() < MAX_SIZE_IN_BYTES, INVALID_FUNCTION_ARGUMENT,
+                "result of repeat function must not take more than 1000000 bytes");
+        return result;
     }
 
-    private static BlockBuilder createBlockBuilder(Type type, long count)
+    private static void checkArguments(long count)
     {
         checkCondition(count <= MAX_RESULT_ENTRIES, INVALID_FUNCTION_ARGUMENT, "count argument of repeat function must be less than or equal to 10000");
```

Comment:
why does it need to be < 10K?

Reply:
I guess this is intentional to avoid generating too large array when this function was added. Since we never heard complaint about it, I think it's reasonable to keep this limit here.